### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by bigfoot90**
Written some unit tests.

Includes https://github.com/Bee-Lab/BeelabTestBundle/pull/4

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,15 +13,23 @@
     ],
     "require": {
         "php": ">=5.4.0",
+        "symfony/doctrine-bridge": "^2.3|^3.0",
+        "symfony/dom-crawler": "^2.3|^3.0",
+        "symfony/browser-kit": "^2.3|^3.0",
         "symfony/framework-bundle": "^2.3|^3.0",
-        "doctrine/data-fixtures": "^1.0"
+        "doctrine/data-fixtures": "^1.0",
+        "mikey179/vfsStream": "^1.6"
```
Comment (garak) [inline composer.json]: vfsStream should be required only in dev, right?

File: phpunit.xml.dist [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    bootstrap                   = "vendor/autoload.php"
+>
+    <testsuites>
+        <testsuite name="Project Test Suite">
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+
+    <php>
+        <server name="KERNEL_DIR" value="app/" />
```
Comment (garak) [inline phpunit.xml.dist]: This is probably wrong, there is no "app/" in a bundle

File: Test/WebTestCase.php [hunk 3]
```diff
@@ -79,22 +80,38 @@ protected function tearDown()
     }
 
     /**
+     * @return ContainerInterface
```
Comment (garak) [inline Test/WebTestCase.php]: Why changing all references to class properties to method calls?

File: Test/WebTestCase.php [hunk 4]
```diff
@@ -79,22 +80,38 @@ protected function tearDown()
...
+
+    /**
      * Save request output and show it in the browser
      * See http://giorgiocefaro.com/blog/test-symfony-and-automatically-open-the-browser-with-the-response-content
      * You can define a "domain" parameter with the current domain of your app.
      *
      * @param bool $delete
      */
-    protected function saveOutput($delete = true)
+    public function saveOutput($delete = true)
```
Comment (garak) [inline Test/WebTestCase.php]: Why changing methods' visibility? If it's only for testing purpose, it's not a great strategy. It's better to change visibility inside tests using reflection.

File: Test/WebTestCase.php (hunk 3 above)
Comment (bigfoot90) [inline Test/WebTestCase.php]: By this way I can mock ```getContainer``` method and return a mock of ```Container``` instead of using the real container.

**Comment by bigfoot90** [comment]
How to  unit test ```commandTest``` method?

File: Test/WebTestCase.php (hunk 3 above)
Comment (garak) [inline Test/WebTestCase.php]: What about mocking `$container` and force it by setting the property? You can override protected properties visibility using reflection.
Another option could be mocking calls to `$kernel::createKernel` and `$kernel->getContainer()`

File: tests/Test/WebTestCaseTest.php [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+    {
+        $mock = $this->getMock('Beelab\TestBundle\Test\WebTestCase', ['getContainer', 'loadFixtureClass']);
+
+        $mock
+            ->method('getContainer')
+            ->willReturn($this->container);
+
+        $mock
+            ->expects($this->exactly(2))
+            ->method('loadFixtureClass');
```
Comment (bigfoot90) [inline tests/Test/WebTestCaseTest.php]: Why this does not work?
The mock calls the original loadFixtureClass method making to fail the test.

File: tests/Test/WebTestCaseTest.php [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+        $this->assertInstanceOf('Beelab\TestBundle\FakeFixtureDependent', $fixture);
+    }
+
+    public function testLoadFixtures()
+    {
+        $this->markTestIncomplete();
+
+        $this->mock
+            ->expects($this->exactly(2))
+            ->method('loadFixtureClass');
```
Comment (bigfoot90) [inline tests/Test/WebTestCaseTest.php]: @garak up
Why this does not work?
The mock calls the original loadFixtureClass method making to fail the test.

File: tests/FakeEntityManager.php [hunk 7]
```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Beelab\TestBundle;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query;
+use Doctrine\ORM\Query\ResultSetMapping;
+
+class FakeEntityManager implements EntityManagerInterface
```
Comment (bigfoot90) [inline tests/FakeEntityManager.php]: Is this the right method of doing this?

File: tests/FakeFixture.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Beelab\TestBundle;
+
+use Doctrine\Common\DataFixtures\FixtureInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class FakeFixture implements FixtureInterface
+{
+    public function load(ObjectManager $manager)
```
Comment (bigfoot90) [inline tests/FakeFixture.php]: Is this the right method of doing this?

File: tests/FakeFixtureDependent.php [hunk 9]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Beelab\TestBundle;
+
+use Doctrine\Common\DataFixtures\DependentFixtureInterface;
+use Doctrine\Common\DataFixtures\FixtureInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class FakeFixtureDependent implements DependentFixtureInterface, FixtureInterface
+{
```
Comment (bigfoot90) [inline tests/FakeFixtureDependent.php]: Is this the right method of doing this?

File: tests/FakeFixture.php [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Beelab\TestBundle;
+
+use Doctrine\Common\DataFixtures\FixtureInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class FakeFixture implements FixtureInterface
+{
```
Comment (bigfoot90) [inline tests/FakeFixture.php]: Is this the right method of doing this?

File: tests/FakeFixtureDependent.php (hunk 9 above)
Comment (garak) [inline tests/FakeFixtureDependent.php]: Looks fine to me

File: tests/FakeFixture.php (hunk 10 above)
Comment (garak) [inline tests/FakeFixture.php]: Same as above

File: Test/WebTestCase.php [hunk 11]
```diff
@@ -341,7 +342,7 @@ private function loadFixtureClass(Loader $loader, $className)
      *
      * @return UploadedFile
      */
-    private function getFile($file, $data, $ext, $mime)
+    protected function getFile($file, $data, $ext, $mime)
```
Comment (garak) [inline Test/WebTestCase.php]: Why this change of visibility?

Comment (bigfoot90) [inline Test/WebTestCase.php]: This bundle should be more generic as possible.
Anyone should be able to generate any kind of file, and you cannot foresee a method for every possible type of file.

Comment (garak) [inline Test/WebTestCase.php]: Seems legit

**Comment by garak** [comment]
Since I merged other PRs, you need to rebase this PR

**Comment by bigfoot90** [comment]
Rebased.
Sorry for the long time past.

File: Test/WebTestCase.php [hunk 12]
```diff
@@ -342,6 +370,11 @@ private function loadFixtureClass(Loader $loader, string $className)
             return;
         }
         $loader->addFixture($fixture);
+        if ($fixture instanceof DependentFixtureInterface) {
```
Comment (garak) [inline Test/WebTestCase.php]: This was removed in 65ec0c4c847640779549edc88043f06ad74d3211
Why adding it again?

Comment (bigfoot90) [inline Test/WebTestCase.php]: It's a merge mistake, corrected.
